simplify-basedomain-usage/README.md

```diff
@@ -0,0 +1,175 @@
+
+We have made the `baseDomain` value available everywhere, and we have lost track of which apps depend on it or how they are using it.
+It's global value that is passed around magically. Changing the `baseDomain` has unknown and potentially catastrophic consequences.
+
+### Complex setup
+
+We configure the `baseDomain` differently depending on how the app that needs the value is deployed:
+
+- For apps that are part of [the app collections deployed to our Management Clusters](https://github.com/giantswarm/capa-app-collection), we pass [the `baseDomain` using the `config` repo](https://github.com/giantswarm/config/blob/main/default/apps/nginx-ingress-controller-app/configmap-values.yaml.template#L2).
+- For [the default apps](https://github.com/giantswarm/default-apps-aws) (apps deployed to both Management Clusters and Workload Clusters), we store the `baseDomain` in a special configmap called `$clustername-cluster-values` created by `cluster-apps-operator` in CAPI installations and by `cluster-operator` in `vintage`. This configmap [is referenced from the `App` CR of the default apps](https://github.com/giantswarm/default-apps-aws/blob/master/helm/default-apps-aws/templates/apps.yaml#L38-L40).
+- For other apps, we also store the `baseDomain` [in several app catalogs configuration](https://github.com/giantswarm/installations/blob/master/gohan/appcatalog/default-appcatalog-values.yaml#L5). That way the `baseDomain` is passed to all applications installed from those catalogs (Honeybadger has already stated that app catalog configuration may go away in the future).
+
+### Differences between `vintage` and CAPI installations
+
+The difference in the value for `baseDomain` between `vintage` and CAPI installations is sometimes confusing while working on applications that will be deployed to both kinds of installations.
+The fact that we reuse the same variable name for values that could potentially contain different things (does it contain info about the management cluster or not?) is confusing and error-prone, adding complexity where this value is used.
+
+### Different clusters in the same installation can't use different `baseDomain` values
+
+We have several controllers reconciling workload clusters, and these controllers receive the `baseDomain` configuration value to create resources for the reconciled clusters.
+Because we pass the `baseDomain` as a parameter to the controllers, all clusters in a single installation must share the same `baseDomain`. There is no way to use different a `baseDomain` for two workload clusters on the same installation.
+
+### Same variable name for different values
+
+In our `installations` repository we save the `baseDomain` in the app catalog, so that this value is automatically passed to all apps from that catalog. For example, [on `goat` we have this value](https://github.com/giantswarm/installations/blob/master/gohan/appcatalog/default-appcatalog-values.yaml#L5)
+```yaml
+baseDomain: gtest.gigantic.io
```

Comment:
Somewhat related, maybe: the `managementCluster` key is also specified there – doesn't that one present similar issues as `baseDomain`? For example, aws-network-topology-operator takes the MC name as argument.